controllers/object_controls.go

```diff
@@ -1344,15 +1350,41 @@ func transformForRuntime(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec,
 		configEnvvarName = "CRIO_CONFIG"
 	}
 
-	setContainerEnv(mainContainer, "RUNTIME_CONFIG", DefaultRuntimeConfigTargetDir+sourceConfigFileName)
-	setContainerEnv(mainContainer, configEnvvarName, DefaultRuntimeConfigTargetDir+sourceConfigFileName)
+	// Handle the top-level configs
+	if runtimeConfigFiles.topLevelConfigFile != "" {
+		sourceConfigFileName := path.Base(runtimeConfigFiles.topLevelConfigFile)
+		sourceConfigDir := path.Dir(runtimeConfigFiles.topLevelConfigFile)
+		setContainerEnv(mainContainer, "RUNTIME_CONFIG", DefaultRuntimeConfigTargetDir+sourceConfigFileName)
+		setContainerEnv(mainContainer, configEnvvarName, DefaultRuntimeConfigTargetDir+sourceConfigFileName)
+
+		volMountConfigName := fmt.Sprintf("%s-config", runtime)
+		volMountConfig := corev1.VolumeMount{Name: volMountConfigName, MountPath: DefaultRuntimeConfigTargetDir}
+		mainContainer.VolumeMounts = append(mainContainer.VolumeMounts, volMountConfig)
+
+		configVol := corev1.Volume{Name: volMountConfigName, VolumeSource: corev1.VolumeSource{HostPath: &corev1.HostPathVolumeSource{Path: sourceConfigDir, Type: newHostPathType(corev1.HostPathDirectoryOrCreate)}}}
+		obj.Spec.Template.Spec.Volumes = append(obj.Spec.Template.Spec.Volumes, configVol)
+	}
+
+	// Handle the drop-in configs
+	if runtimeConfigFiles.dropInConfigFile != "" {
+		sourceConfigFileName := path.Base(runtimeConfigFiles.dropInConfigFile)
+		sourceConfigDir := path.Dir(runtimeConfigFiles.dropInConfigFile)
```

Comment:
Why sourceConfig* if these are the drop-in file/folder names, shouldn't this be `dropinConfigDir` and `dropinConfigFileName` ?

Reply:
This doesn't actually matter. The variables are scoped to the conditional block. This also calls out that we may have some comonality here so that we *could* factor this out into a function.